### ISSUE TITLE
Visual Studios New Map

### ISSUE THREAD

**Issue by DestinyFrank** [lgsvl/simulator#1925]
I am learning to create a simulation using the Visual Editor. I want to use a new map from my library that I created and uploaded to the website. Is there a way to add the new to the Maps in the Editor? 

![image](https://user-images.githubusercontent.com/86273840/146826923-b517f086-b1eb-4e2c-b1ea-97ee1ef413c1.png)

**Comment by EricBoiseLGSVL** [comment]
Sure, VSE will have access if you add to your library.  I always run a simulation with the map locally from the webui and then VSE has the map cached.

**Comment by DestinyFrank** [comment]
I have added the new map to my Libary and run the map in simulation. It was still not showing in the VSE options.

**Comment by EricBoiseLGSVL** [comment]
No, unless you set data folder to a different path.  What data path is the exe and editor on?

**Comment by DestinyFrank** [comment]
Do you mean it needs to be in the exact file location? I apologize; I am unsure about the data path.

**Comment by EricBoiseLGSVL** [comment]
No worries.  So we have a config.yml file we use to specify a data path if users don't want to use the default.  If you don't then the data path is set to application.presistentdatapath from Unity apis.  So it would be like so.
![image](https://user-images.githubusercontent.com/45106230/146980978-e5990158-fe27-41b6-91b4-35eebc09ef1f.png)

This is were you cache the assets you pull from the webui.

My concern is that you are running the exe and editor to different data paths so VSE will not have it cached.  If you haven't altered a config then the exe and unity editor should be on the same path.

The next thing to check if the map is public.  Try with a test map to be sure or an existing map we provide.  Last would be to clear cache of maps with the cache control menu in the main menu.  Clear all local in case you have some name conflict and try to run the sim again, then vse

**Comment by DestinyFrank** [comment]
Here is my folder with the simulator is located. I am unsure if this is correct or not. My map is private, but my other private maps show in the VSE already. 

Also, would I clear the cache in the VSE or the website? 

![image](https://user-images.githubusercontent.com/86273840/147091176-5a119ebe-4a79-4957-824e-59cc6a97ab56.png)

**Comment by EricBoiseLGSVL** [comment]
No this is not correct.
Simulator repo should be separate from the built binary.
Do not build the binary inside the repository source.
SVL Source:
![image](https://user-images.githubusercontent.com/45106230/147136769-9e2d67a7-5573-475f-a085-eed625599049.png)

Simulator asset source in simulator repo:
![image](https://user-images.githubusercontent.com/45106230/147136881-1acfce15-9f1c-49a4-a4b6-b9288e81b83e.png)

Simulator appdata folder with cached files and database:
![image](https://user-images.githubusercontent.com/45106230/147136967-43c4715a-d758-42e3-87e9-fc43a4f88c11.png)

Simulator Build:
![image](https://user-images.githubusercontent.com/45106230/147137058-70d9351c-e445-4ddd-9b87-b93a17c5d47b.png)

Run the unity editor simulator (after you remove build files) and start with LoaderScene.scene in Assets/Scenes/LoaderScene
Only run binary once you are done developing and want to run end to end tests with AD stack to have better performance.

**Comment by EricBoiseLGSVL** [comment]
When I clear cache after large changes in source I have a few steps.
1- I remove all assets in WebUI that relate to changes. Start fresh to make sure all cached assets are latest
2 - I delete cached asset in simulator, run loaderscene, go to drop down menu top right and select cache control, delete cached assets.  You can also just delete the entire appdata folder and relink the cluster.
3 - I build new assets and upload to webui.
4 - Create new simulations that use the new assets in webui.
Then unity editor will have all new cached assets.
